File: root/cgi-bin/hello.php
<?php
header("Content-Type: text/plain");

echo "Hello CGI via PHP!\n";
echo "Method: " . $_SERVER['REQUEST_METHOD'] . "\n";
echo "Query: " . (isset($_SERVER['QUERY_STRING']) ? $_SERVER['QUERY_STRING'] : '') . "\n";
echo "Body-Len: " . strlen(file_get_contents("php://input")) . "\n";
?>

File: root/cgi-bin/hello.py
#!/usr/bin/env python3
import os, sys
print("Hello CGI! Method:", os.environ.get("REQUEST_METHOD"))
print("Query:", os.environ.get("QUERY_STRING", ""))
print("Body-len:", os.environ.get("CONTENT_LENGTH", "0"))
open("local.txt", "w").write("ok")
